guava/src/com/google/common/io/ReaderInputStream.java

```diff
@@ -185,6 +185,76 @@ public int read(byte[] b, int off, int len) throws IOException {
     }
   }
 
+  @Override
+  public int available() throws IOException {
+    if (byteBuffer.hasRemaining()) {
+      // if we have remaining chars in the buffer, return their count
+      return byteBuffer.remaining();
+    } else if (charBuffer.hasRemaining()) {
+      // if not, try to make quick progress by encoding available
+      // chars that have been read from the users reader
+      CoderResult result = encoder.encode(charBuffer, byteBuffer, endOfInput);
+
+      if (result.isOverflow()) {
+        // the byte buffer is too small, enlarge it
+        startDraining(true);
+      } else if (result.isUnderflow()) {
+        // the char buffer is too small, enlarge it
+        charBuffer = grow(charBuffer);
+      } else if (result.isError()) {
+        // Only reach here if a CharsetEncoder with non-REPLACE settings is used.
```

Comment:
```suggestion
        // Only reach here if a CharsetEncoder with REPORT action is used.
```
[`IGNORE`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/charset/CodingErrorAction.html#IGNORE) would not cause this either.